api/model/src/main/java/org/projectnessie/model/Content.java

```diff
@@ -97,6 +97,8 @@ public interface Type {
   @JsonIgnore
   public abstract Type getType();
 
+  public abstract Content withId(String id);
```

Comment:
Maybe:

```suggestion
  public abstract Content withId(@Nullable @jakarta.annotation.Nullable String id);
```

Conversely we could require `id` to be non null and introduce a `withoutId()` method, wdyt?

Reply:
Nope - ID must be nullable - otherwise you could never add new content